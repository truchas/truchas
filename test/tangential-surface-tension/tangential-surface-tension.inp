&MESH
  mesh_file = 'tangential-surface-tension.gen'
/

&OUTPUTS
  Output_T = 0.0, 10.0
  !Output_T = 0.0, 2e-3
  !Output_T = 0.0, 1.0
  Output_Dt = 5.0 !1.0
/

&PHYSICS
  heat_transport = .true.
  flow = .true.
/

&DIFFUSION_SOLVER
  abs_temp_tol       = 0.0
  rel_temp_tol       = 1.0e-3
  abs_enthalpy_tol   = 0.0
  rel_enthalpy_tol   = 1.0e-3
  nlk_tol            = 0.01
  nlk_preconditioner = 'hypre_amg'
  pc_amg_cycles      = 2
  verbose_stepping   = .true.
/

&FLOW
  inviscid = f
  courant_number = 0.5
  viscous_number = 10.0
  viscous_implicitness = 0.5
/

&FLOW_VISCOUS_SOLVER
  rel_tol = 1.0e-10
  abs_tol = 1.0e-10
  max_ds_iter = 50
  max_amg_iter = 10
  krylov_method = 'cg'
/ 

&FLOW_PRESSURE_SOLVER
  rel_tol = 1.0e-10
  abs_tol = 1.0e-10
  max_ds_iter = 100
  max_amg_iter = 10
  krylov_method = 'cg'
/

&NUMERICS
  dt_init = 1.0d-03
  dt_min  = 1.0d-10
  dt_grow = 1.5
  dt_max  = 1.0
  discrete_ops_type = 'ortho'
/

##### BOUNDARY CONDITIONS ######################################################

&DS_BOUNDARY_CONDITION
  name = 'cold surface'
  variable = 'temperature'
  condition = 'dirichlet'
  face_set_ids = 1, 2, 4
  data_constant = 1.0
/

&DS_BOUNDARY_CONDITION
  name = 'hot top'
  variable = 'temperature'
  condition = 'dirichlet'
  face_set_ids = 3
  data_constant = 2.0
/

&DS_BOUNDARY_CONDITION
  name = 'x-z symmetry'
  variable = 'temperature'
  condition = 'flux'
  face_set_ids = 5, 6
  data_constant = 0.0
/

&FLOW_BC
  name = 'surface tension'
  type = 'marangoni'
  face_set_ids = 3
  dsigma = -1.0
/

&FLOW_BC
  name = 'walls'
  type = 'free-slip'
  face_set_ids = 1, 2, 4, 5, 6
/

&BODY
  surface_name = 'from mesh file'
  mesh_material_number = 1
  material_number = 1
  temperature = 1
/

##### MATERIALS ################################################################

&MATERIAL
  material_name = 'liquid'
  material_number = 1
  material_feature = 'background'
  immobile= .false.
  priority = 1
  density = 1.0 ! not void
/

&PHASE 
 name = 'liquid' 
 property_name(1) = 'density',       property_constant(1) = 1.0
 property_name(2) = 'specific heat', property_constant(2) = 1.0
 property_name(3) = 'conductivity',  property_constant(3) = 1.0
 property_name(4) = 'viscosity',     property_constant(4) = 1.0
/

&MATERIAL_SYSTEM
  name = 'single'
  phases = 'liquid'
/
