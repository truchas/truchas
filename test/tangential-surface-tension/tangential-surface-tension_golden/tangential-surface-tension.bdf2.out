
TRAP step      1: T= 0.00000E+00, H= 1.00000E-03, ETAH= 5.00000E-04
  Preconditioner updated at T= 1.00000E-03, ETAH= 5.00000E-04
    1: error= 1.61091E-03
    2: error= 6.81998E-07
  AIN BCE solve succeeded:   2 iterations, error= 6.81998E-07
  Step accepted: no local error control

BDF2 step      2: T= 1.00000E-03, H= 1.00000E-03, ETAH= 6.66667E-04
    1: error= 2.03332E-04
    2: error= 1.20707E-06
  AIN BCE solve succeeded:   2 iterations, error= 1.20707E-06
  Step accepted: no local error control

BDF2 step      3: T= 2.00000E-03, H= 1.00000E-03, ETAH= 6.66667E-04
    1: error= 6.92379E-06
  AIN BCE solve succeeded:   1 iterations, error= 6.92379E-06
  Step accepted: no local error control

BDF2 step      4: T= 3.00000E-03, H= 1.00000E-03, ETAH= 6.66667E-04
    1: error= 9.67942E-06
  AIN BCE solve succeeded:   1 iterations, error= 9.67942E-06
  Step accepted: perr= 9.67942E-06

BDF2 step      5: T= 4.00000E-03, H= 1.50000E-03, ETAH= 9.37500E-04
    1: error= 1.60198E-05
  AIN BCE solve succeeded:   1 iterations, error= 1.60198E-05
  Step accepted: perr= 1.60198E-05

BDF2 step      6: T= 5.50000E-03, H= 2.25000E-03, ETAH= 1.40625E-03
    1: error= 3.82392E-05
  AIN BCE solve succeeded:   1 iterations, error= 3.82392E-05
  Step accepted: perr= 3.82392E-05

BDF2 step      7: T= 7.75000E-03, H= 3.37500E-03, ETAH= 2.10937E-03
  Preconditioner updated at T= 1.11250E-02, ETAH= 2.10937E-03
    1: error= 4.19790E-04
    2: error= 9.48073E-08
  AIN BCE solve succeeded:   2 iterations, error= 9.48073E-08
  Step accepted: perr= 4.19790E-04

BDF2 step      8: T= 1.11250E-02, H= 5.06250E-03, ETAH= 3.16406E-03
    1: error= 4.16427E-04
    2: error= 2.06279E-04
  AIN BCE solve succeeded:   2 iterations, error= 2.06279E-04
  Step accepted: perr= 6.22704E-04

BDF2 step      9: T= 1.61875E-02, H= 7.59375E-03, ETAH= 4.74609E-03
    1: error= 1.06859E-03
    2: error= 1.31804E-03
  AIN BCE solve succeeded:   2 iterations, error= 1.31804E-03
  Step accepted: perr= 2.38662E-03

BDF2 step     10: T= 2.37812E-02, H= 1.13906E-02, ETAH= 7.11914E-03
    1: error= 2.38822E-03
    2: error= 5.56374E-03
  AIN BCE solve succeeded:   2 iterations, error= 5.56374E-03
  Step accepted: perr= 7.95194E-03

BDF2 step     11: T= 3.51719E-02, H= 1.70859E-02, ETAH= 1.06787E-02
  Preconditioner updated at T= 5.22578E-02, ETAH= 1.06787E-02
    1: error= 2.54310E-02
    2: error= 6.73228E-06
  AIN BCE solve succeeded:   2 iterations, error= 6.73228E-06
  Step accepted: perr= 2.54310E-02

BDF2 step     12: T= 5.22578E-02, H= 2.56289E-02, ETAH= 1.60181E-02
    1: error= 5.38842E-02
    2: error= 2.57261E-02
    3: error= 8.22040E-05
  AIN BCE solve succeeded:   3 iterations, error= 8.22040E-05
  Step accepted: perr= 7.95260E-02

BDF2 step     13: T= 7.78867E-02, H= 3.84434E-02, ETAH= 2.40271E-02
    1: error= 1.10744E-01
    2: error= 1.29692E-01
    3: error= 1.01655E-03
  AIN BCE solve succeeded:   3 iterations, error= 1.01655E-03
  Step accepted: perr= 2.39404E-01

BDF2 step     14: T= 1.16330E-01, H= 4.54446E-02, ETAH= 2.94764E-02
    1: error= 1.55853E-01
    2: error= 2.53893E-01
    3: error= 2.81706E-03
  AIN BCE solve succeeded:   3 iterations, error= 2.81706E-03
  Step accepted: perr= 4.06889E-01

BDF2 step     15: T= 1.61775E-01, H= 4.44609E-02, ETAH= 2.97491E-02
    1: error= 1.55948E-01
    2: error= 2.57777E-01
    3: error= 2.99550E-03
  AIN BCE solve succeeded:   3 iterations, error= 2.99550E-03
  Step accepted: perr= 4.10687E-01

BDF2 step     16: T= 2.06236E-01, H= 4.85251E-02, ETAH= 3.18855E-02
    1: error= 1.63935E-01
    2: error= 3.00141E-01
    3: error= 3.97778E-03
  AIN BCE solve succeeded:   3 iterations, error= 3.97778E-03
  Step accepted: perr= 4.60049E-01

BDF2 step     17: T= 2.54761E-01, H= 4.90137E-02, ETAH= 3.26213E-02
    1: error= 1.47754E-01
    2: error= 2.79754E-01
    3: error= 3.95879E-03
  AIN BCE solve succeeded:   3 iterations, error= 3.95879E-03
  Step accepted: perr= 4.23507E-01

BDF2 step     18: T= 3.03774E-01, H= 5.25650E-02, ETAH= 3.46397E-02
    1: error= 1.42915E-01
    2: error= 2.94524E-01
    3: error= 4.66771E-03
  AIN BCE solve succeeded:   3 iterations, error= 4.66771E-03
  Step accepted: perr= 4.32729E-01

BDF2 step     19: T= 3.56339E-01, H= 5.49918E-02, ETAH= 3.63875E-02
    1: error= 1.32005E-01
    2: error= 2.91179E-01
    3: error= 5.06831E-03
  AIN BCE solve succeeded:   3 iterations, error= 5.06831E-03
  Step accepted: perr= 4.18077E-01

BDF2 step     20: T= 4.11331E-01, H= 5.86431E-02, ETAH= 3.86811E-02
    1: error= 1.25233E-01
    2: error= 2.99865E-01
    3: error= 5.79393E-03
  AIN BCE solve succeeded:   3 iterations, error= 5.79393E-03
  Step accepted: perr= 4.19267E-01

BDF2 step     21: T= 4.69974E-01, H= 6.21786E-02, ETAH= 4.10520E-02
    1: error= 1.17225E-01
    2: error= 3.03443E-01
    3: error= 6.45485E-03
  AIN BCE solve succeeded:   3 iterations, error= 6.45485E-03
  Step accepted: perr= 4.14178E-01

BDF2 step     22: T= 5.32153E-01, H= 6.63125E-02, ETAH= 4.37393E-02
  Preconditioner updated at T= 5.98465E-01, ETAH= 4.37393E-02
    1: error= 4.12317E-01
    2: error= 2.58731E-04
  AIN BCE solve succeeded:   2 iterations, error= 2.58731E-04
  Step accepted: perr= 4.12317E-01

BDF2 step     23: T= 5.98465E-01, H= 7.07793E-02, ETAH= 4.66792E-02
    1: error= 3.85746E-01
    2: error= 2.29342E-02
    3: error= 4.75378E-05
  AIN BCE solve succeeded:   3 iterations, error= 4.75378E-05
  Step accepted: perr= 4.08579E-01

BDF2 step     24: T= 6.69245E-01, H= 7.58144E-02, ETAH= 4.99708E-02
    1: error= 3.60300E-01
    2: error= 4.51562E-02
    3: error= 1.92201E-04
  AIN BCE solve succeeded:   3 iterations, error= 1.92201E-04
  Step accepted: perr= 4.05203E-01

BDF2 step     25: T= 7.45059E-01, H= 8.14313E-02, ETAH= 5.36488E-02
    1: error= 3.35145E-01
    2: error= 6.64511E-02
    3: error= 4.42128E-04
  AIN BCE solve succeeded:   3 iterations, error= 4.42128E-04
  Step accepted: perr= 4.01086E-01

BDF2 step     26: T= 8.26490E-01, H= 8.77852E-02, ETAH= 5.78001E-02
    1: error= 3.10644E-01
    2: error= 8.69244E-02
    3: error= 8.06641E-04
  AIN BCE solve succeeded:   3 iterations, error= 8.06641E-04
  Step accepted: perr= 3.96689E-01

BDF2 step     27: T= 9.14276E-01, H= 9.49978E-02, ETAH= 6.25097E-02
    1: error= 2.86731E-01
    2: error= 1.06486E-01
    3: error= 1.29254E-03
  AIN BCE solve succeeded:   3 iterations, error= 1.29254E-03
  Step accepted: perr= 3.91847E-01

BDF2 step     28: T= 1.00927E+00, H= 1.03243E-01, ETAH= 6.78874E-02
    1: error= 2.72240E-01
    2: error= 1.33168E-01
    3: error= 1.90576E-03
  AIN BCE solve succeeded:   3 iterations, error= 1.90576E-03
  Step accepted: perr= 4.05106E-01

BDF2 step     29: T= 1.11252E+00, H= 1.09960E-01, ETAH= 7.25447E-02
    1: error= 2.57208E-01
    2: error= 1.49083E-01
    3: error= 2.32839E-03
  AIN BCE solve succeeded:   3 iterations, error= 2.32839E-03
  Step accepted: perr= 4.05816E-01

BDF2 step     30: T= 1.22248E+00, H= 1.17952E-01, ETAH= 7.77259E-02
    1: error= 2.41261E-01
    2: error= 1.63875E-01
    3: error= 2.75321E-03
  AIN BCE solve succeeded:   3 iterations, error= 2.75321E-03
  Step accepted: perr= 4.04430E-01

BDF2 step     31: T= 1.34043E+00, H= 1.26785E-01, ETAH= 8.35188E-02
    1: error= 2.23692E-01
    2: error= 1.77101E-01
    3: error= 3.13884E-03
  AIN BCE solve succeeded:   3 iterations, error= 3.13884E-03
  Step accepted: perr= 3.99897E-01

BDF2 step     32: T= 1.46721E+00, H= 1.36817E-01, ETAH= 9.00691E-02
    1: error= 2.09505E-01
    2: error= 1.92487E-01
    3: error= 3.90561E-03
  AIN BCE solve succeeded:   3 iterations, error= 3.90561E-03
  Step accepted: perr= 4.01823E-01

BDF2 step     33: T= 1.60403E+00, H= 1.46965E-01, ETAH= 9.68225E-02
    1: error= 1.93420E-01
    2: error= 2.02153E-01
    3: error= 4.79973E-03
  AIN BCE solve succeeded:   3 iterations, error= 4.79973E-03
  Step accepted: perr= 3.95101E-01

BDF2 step     34: T= 1.75100E+00, H= 1.59505E-01, ETAH= 1.04906E-01
    1: error= 1.78806E-01
    2: error= 2.13613E-01
    3: error= 5.91052E-03
  AIN BCE solve succeeded:   3 iterations, error= 5.91052E-03
  Step accepted: perr= 3.91495E-01

BDF2 step     35: T= 1.91050E+00, H= 1.73315E-01, ETAH= 1.13967E-01
    1: error= 1.63986E-01
    2: error= 2.23700E-01
    3: error= 7.10605E-03
  AIN BCE solve succeeded:   3 iterations, error= 7.10605E-03
  Step accepted: perr= 3.86881E-01

BDF2 step     36: T= 2.08381E+00, H= 1.89096E-01, ETAH= 1.24260E-01
    1: error= 1.50050E-01
    2: error= 2.32833E-01
    3: error= 8.47707E-03
  AIN BCE solve succeeded:   3 iterations, error= 8.47707E-03
  Step accepted: perr= 3.81368E-01

BDF2 step     37: T= 2.27291E+00, H= 2.07484E-01, ETAH= 1.36217E-01
    1: error= 1.36388E-01
    2: error= 2.41349E-01
    3: error= 9.88187E-03
  AIN BCE solve succeeded:   3 iterations, error= 9.88187E-03
  Step accepted: perr= 3.75593E-01

BDF2 step     38: T= 2.48039E+00, H= 2.28798E-01, ETAH= 1.50088E-01
    1: error= 1.23421E-01
    2: error= 2.49397E-01
    3: error= 1.11760E-02
    4: error= 3.22228E-04
  AIN BCE solve succeeded:   4 iterations, error= 3.22228E-04
  Step accepted: perr= 3.69513E-01

BDF2 step     39: T= 2.70919E+00, H= 2.53715E-01, ETAH= 1.66281E-01
    1: error= 1.10764E-01
    2: error= 2.56004E-01
    3: error= 1.23967E-02
    4: error= 3.75953E-04
  AIN BCE solve succeeded:   4 iterations, error= 3.75953E-04
  Step accepted: perr= 3.62243E-01

BDF2 step     40: T= 2.96291E+00, H= 2.83409E-01, ETAH= 1.85520E-01
  Preconditioner updated at T= 3.24632E+00, ETAH= 1.85520E-01
    1: error= 3.55201E-01
    2: error= 1.21639E-03
  AIN BCE solve succeeded:   2 iterations, error= 1.21639E-03
  Step accepted: perr= 3.55476E-01

BDF2 step     41: T= 3.24632E+00, H= 3.18464E-01, ETAH= 2.08266E-01
    1: error= 3.57510E-01
    2: error= 3.88466E-02
    3: error= 2.83015E-04
  AIN BCE solve succeeded:   3 iterations, error= 2.83015E-04
  Step accepted: perr= 3.95823E-01

BDF2 step     42: T= 3.56478E+00, H= 3.07500E-01, ETAH= 2.06204E-01
    1: error= 2.93293E-01
    2: error= 2.90874E-02
    3: error= 1.75340E-04
  AIN BCE solve succeeded:   3 iterations, error= 1.75340E-04
  Step accepted: perr= 3.21886E-01

BDF2 step     43: T= 3.87228E+00, H= 2.96914E-01, ETAH= 1.99105E-01
    1: error= 2.38121E-01
    2: error= 1.59442E-02
    3: error= 5.47064E-05
  AIN BCE solve succeeded:   3 iterations, error= 5.47064E-05
  Step accepted: perr= 2.53477E-01

BDF2 step     44: T= 4.16919E+00, H= 2.86692E-01, ETAH= 1.92250E-01
    1: error= 1.82733E-01
    2: error= 6.45199E-03
  AIN BCE solve succeeded:   2 iterations, error= 6.45199E-03
  Step accepted: perr= 1.88722E-01

BDF2 step     45: T= 4.45589E+00, H= 2.76822E-01, ETAH= 1.85632E-01
    1: error= 1.38547E-01
    2: error= 5.67494E-04
  AIN BCE solve succeeded:   2 iterations, error= 5.67494E-04
  Step accepted: perr= 1.38744E-01

BDF2 step     46: T= 4.73271E+00, H= 2.67292E-01, ETAH= 1.79241E-01
    1: error= 1.05307E-01
    2: error= 3.09882E-03
  AIN BCE solve succeeded:   2 iterations, error= 3.09882E-03
  Step accepted: perr= 1.02209E-01

BDF2 step     47: T= 5.00000E+00, H= 4.00938E-01, ETAH= 2.50586E-01
    1: error= 1.50287E-01
    2: error= 4.61359E-02
    3: error= 7.17314E-04
  AIN BCE solve succeeded:   3 iterations, error= 7.17314E-04
  Step accepted: perr= 1.96007E-01

BDF2 step     48: T= 5.40094E+00, H= 5.53698E-01, ETAH= 3.50440E-01
    1: error= 1.85857E-01
    2: error= 1.37834E-01
    3: error= 5.12049E-03
  AIN BCE solve succeeded:   3 iterations, error= 5.12049E-03
  Step accepted: perr= 3.22856E-01

BDF2 step     49: T= 5.95464E+00, H= 5.77310E-01, ETAH= 3.82213E-01
    1: error= 1.79763E-01
    2: error= 1.58222E-01
    3: error= 6.68405E-03
  AIN BCE solve succeeded:   3 iterations, error= 6.68405E-03
  Step accepted: perr= 3.37107E-01

BDF2 step     50: T= 6.53194E+00, H= 6.64968E-01, ETAH= 4.33124E-01
    1: error= 2.06834E-01
    2: error= 2.33125E-01
    3: error= 1.14476E-02
    4: error= 1.17886E-03
  AIN BCE solve succeeded:   4 iterations, error= 1.17886E-03
  Step accepted: perr= 4.44381E-01

BDF2 step     51: T= 7.19691E+00, H= 6.27483E-01, ETAH= 4.22406E-01
    1: error= 1.58005E-01
    2: error= 1.72861E-01
    3: error= 8.22395E-03
  AIN BCE solve succeeded:   3 iterations, error= 8.22395E-03
  Step accepted: perr= 3.32441E-01

BDF2 step     52: T= 7.82440E+00, H= 5.92112E-01, ETAH= 3.98595E-01
    1: error= 1.27548E-01
    2: error= 1.28117E-01
    3: error= 5.10071E-03
  AIN BCE solve succeeded:   3 iterations, error= 5.10071E-03
  Step accepted: perr= 2.55672E-01

BDF2 step     53: T= 8.41651E+00, H= 5.58735E-01, ETAH= 3.76126E-01
    1: error= 9.23661E-02
    2: error= 8.44255E-02
    3: error= 2.97781E-03
  AIN BCE solve succeeded:   3 iterations, error= 2.97781E-03
  Step accepted: perr= 1.76344E-01

BDF2 step     54: T= 8.97524E+00, H= 5.27239E-01, ETAH= 3.54924E-01
    1: error= 6.61465E-02
    2: error= 5.45718E-02
    3: error= 1.59574E-03
  AIN BCE solve succeeded:   3 iterations, error= 1.59574E-03
  Step accepted: perr= 1.20263E-01

BDF2 step     55: T= 9.50248E+00, H= 4.97518E-01, ETAH= 3.34917E-01
    1: error= 4.77434E-02
    2: error= 3.54765E-02
    3: error= 8.05036E-04
  AIN BCE solve succeeded:   3 iterations, error= 8.05036E-04
  Step accepted: perr= 8.29202E-02
